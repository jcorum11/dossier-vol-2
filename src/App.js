import About from "./pages/About";
import Contact from "./pages/Contact";
import Nav from "./components/Nav";
import Projects from "./pages/Projects";
import Resume from "./pages/Resume";
import styled from 'styled-components';
import Landing from './pages/Landing'
import { pink, blue } from './variables'
import { useEffect, useRef, useState } from 'react'

function App() {
  const [currentSectionOption, setCurrentSectionOption] = useState('landing')
  const sectionOptions = ['landing', 'about', 'projects', 'resume', 'contact']
  let sectionOptionIndex = 0
  const currentSection = useRef(null)
  useEffect(() => {
    window.addEventListener('scroll', handleScroll)
  }, [])
  const handleScroll = () => {
    if (currentSectionOption === currentSectionOption[currentSectionOption.length - 1]) {
      sectionOptionIndex = 0
      setCurrentSectionOption('landing')
    } else {
      sectionOptionIndex++
      setCurrentSectionOption(sectionOptions[sectionOptionIndex])
    }
    currentSection.current.scrollIntoView({behavior: 'smooth'})
  }
  return (
    <Container>
      <Nav />
      <Body>
        <Landing />
        <About currentSection={currentSection} currentSectionOption={currentSectionOption} />
        <Projects />
        <Resume />
        <Contact />
      </Body>
      <footer>
        <FooterContent>
          <Paragraph>Designed by Kyanna Corum <span role='img' aria-label='heart'>❤️ </span></Paragraph>
          <Paragraph>Developed by Jake <span role="img" aria-label="peace sign">✌</span></Paragraph>
        </FooterContent>
      </footer>
    </Container>
  );
}

const Container = styled.div`
min-height: 100vh;
`

const FooterContent = styled.div`
color: ${pink};
text-align: center;
background-color: ${blue};
padding: 1rem;
`

const Paragraph = styled.p`
margin: 0;
`

const Body = styled.section`
scroll-snap-type: y mandatory;
`

export default App;
