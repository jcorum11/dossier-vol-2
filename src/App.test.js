import { render, cleanup } from '@testing-library/react'
import {lazy, Suspense} from 'react'
import App from './App'

afterEach(cleanup)

it('renders', async () => {
  render(
  <Suspense fallback={<div>loading...</div>}>
    <App />
  </Suspense>
  )
})

it('matches snapshot', async () => {
  const { asFragment } = render(
  <Suspense fallback={<div>loading...</div>}>
    <App />
  </Suspense>)
  expect(asFragment()).toMatchSnapshot()
})
