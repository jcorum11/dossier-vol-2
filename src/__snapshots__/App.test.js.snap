// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL ewLFSZ"
    style="display: none;"
  >
    <nav
      class="sc-hKwDye kUPOum"
    >
      <div
        class="sc-iCfMLu jhzBdE"
      >
        <div
          class="sc-furwcr hWpAUk"
        >
          <a
            class="sc-eCImPb bwVhyj"
            href="#about"
          >
            About Me
          </a>
        </div>
        <div
          class="sc-furwcr hWpAUk"
        >
          <a
            class="sc-eCImPb bwVhyj"
            href="#projects"
          >
            Projects
          </a>
        </div>
        <div
          class="sc-furwcr hWpAUk"
        >
          <a
            class="sc-eCImPb bwVhyj"
            href="#resume"
          >
            Resume
          </a>
        </div>
        <div
          class="sc-furwcr hWpAUk"
        >
          <a
            class="sc-eCImPb bwVhyj"
            href="#contact"
          >
            Contact Me
          </a>
        </div>
      </div>
    </nav>
    <main>
      <section
        class="sc-pVTFL havwbf"
        id="about"
      >
        <div
          class="sc-jrQzAO fptzda"
        >
          <div
            class="sc-crHmcD jtzwLr"
          >
            <div
              class="sc-iqseJM bARYZD"
            />
            <div
              class="sc-ksdxgE eoUvfN"
            >
              <p
                class="sc-kDTinF cbTbHh"
              >
                I have worked to overcome obstacles to become a full stack developer graduating from University of Utah’s boot camp near the top of the class, earning me a full-stack developer certificate with a background in Industrial/Organizational Psychology.
              </p>
            </div>
          </div>
          <div
            class="sc-crHmcD jtzwLr"
          >
            <div
              class="sc-ksdxgE sc-hBUSln eoUvfN unhP"
            >
              <p
                class="sc-kDTinF sc-bqiRlB cbTbHh kxIeLj"
              >
                I have studied diligently to learn and implement React, Redux, and GraphQL. I implemented all three of these in a project called Rolodeck where I worked with a team and implemented GraphQL for the entire project, programmed a QR scanner in React, and implemented Redux for easily transferable data.
              </p>
            </div>
            <div
              class="sc-iqseJM sc-egiyK bARYZD jDDiEi"
            />
          </div>
        </div>
      </section>
      <section
        class="py-5"
        id="projects"
      >
        <section />
      </section>
      <section
        id="resume"
      >
        <section
          class="sc-fotOHu hRWjbf"
        >
          <div
            class="sc-bkkeKt fSnNen"
          >
            <h1
              class="sc-fFeiMQ cgaIkV"
            >
              WANT TO HIRE ME?
            </h1>
          </div>
          <div
            class="sc-dlVxhl rozkw"
          >
            <a
              class="sc-ieecCq jyiwkL"
              href="jacob-corum-resume-oct-2020.pdf"
            >
              <div
                class="sc-dJjYzT jSpXhQ"
              />
            </a>
            <a
              class="sc-hGPBjI fhbPZP"
              download="jake-corums-resume"
              href="jacob-corum-resume-oct-2020.pdf"
            >
              download
            </a>
          </div>
        </section>
      </section>
      <section
        class="sc-bYoBSM eNNAei"
        id="contact"
      >
        <section
          class="sc-iwjdpV lcRVHO"
        >
          <div
            class="sc-giYglK fsAClQ"
          >
            <div
              class="sc-llYSUQ bLLUyJ"
            >
              <a
                class="sc-ezbkAF jvdZVz"
                href="href='https://www.linkedin.com/in/jacob-corum-full-stack-development/'"
              >
                <p
                  class="sc-iJKOTD dIsZes"
                >
                  LinkedIn
                </p>
              </a>
            </div>
            <div
              class="sc-llYSUQ bLLUyJ"
            >
              <a
                class="sc-ezbkAF jvdZVz"
                href="https://github.com/jcorum11"
              >
                <p
                  class="sc-iJKOTD dIsZes"
                >
                  Github
                </p>
              </a>
            </div>
          </div>
          <div
            class="sc-cxpSdN JUJVF"
          >
            <a
              class="sc-kfPuZi sc-fKVqWL skVYV bmweRq"
              href="mailto: jacob.w.corum@gmail.com"
            >
              jacob.w.corum@gmail.com
            </a>
            <a
              class="sc-kfPuZi sc-bBHxTw skVYV eFkXRc"
              href="tel: 801-814-3191"
            >
              801-814-3191
            </a>
          </div>
        </section>
      </section>
    </main>
    <footer>
      <div
        class="sc-gsDKAQ dHSMqL"
      >
        <p
          class="sc-dkPtRN kKyqYD"
        >
          Designed by Kyanna Corum 
          <span
            aria-label="heart"
            role="img"
          >
            ❤️ 
          </span>
        </p>
        <p
          class="sc-dkPtRN kKyqYD"
        >
          Developed by Jake 
          <span
            aria-label="peace sign"
            role="img"
          >
            ✌
          </span>
        </p>
      </div>
    </footer>
  </div>
  <div>
    loading...
  </div>
</DocumentFragment>
`;
